homeassistant/components/thethingsnetwork/entity.py

```diff
@@ -0,0 +1,118 @@
+"""Support for The Things Network entities."""
+
+from abc import ABC
+import logging
+from typing import TYPE_CHECKING, Optional
+
+from ttn_client import TTNSensorValue
+
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+if TYPE_CHECKING:
+    from .coordinator import TTNCoordinator
+
+from .const import CONF_APP_ID
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TTNEntity(CoordinatorEntity["TTNCoordinator"], Entity, ABC):
+    """Representation of a The Things Network Data Storage sensor."""
+
+    @staticmethod
+    def get_unique_id(device_id: str, field_id: str) -> str:
+        """Get unique_id which is derived from device_id and field_id."""
+        return f"{device_id}_{field_id}"
+
+    def __init__(
+        self,
+        coordinator: "TTNCoordinator",
+        ttn_value: TTNSensorValue,
+    ) -> None:
+        """Initialize a The Things Network Data Storage sensor."""
+
+        self._entry = coordinator.config_entry
+        self._ttn_value = ttn_value
+        self._name = f"{self.device_name} {self.field_id}"
```

Comment:
New integrations should use `_attr_has_entity_name = True`. Although this is an older integration, it's still a new integration. https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations